Sources/Identity.swift

```diff
@@ -64,20 +64,25 @@ import Foundation
     /// Handles update identity requests to add/update customer identifiers.
     /// - Parameter event: the identity request event
     private func handleUpdateIdentity(event: Event) {
-        state.updateCustomerIdentifiers(event: event, createXDMSharedState: createXDMSharedState(data:event:))
+        // [MOB-16019] Adding pending shared state to avoid race condition between updating and reading identity map
+        let resolver = createPendingXDMSharedState(event: event)
+        state.updateCustomerIdentifiers(event: event, resolveXDMSharedState: resolver)
     }
 
     /// Handles remove identity requests to remove customer identifiers.
     /// - Parameter event: the identity request event
     private func handleRemoveIdentity(event: Event) {
-        state.removeCustomerIdentifiers(event: event, createXDMSharedState: createXDMSharedState(data:event:))
+        // [MOB-16019] Adding pending shared state to avoid race condition between updating and reading identity map
```

Comment:
let's remove the ticket id from this comment, it can be found based on commit history.